File: epi_judge_python/spreadsheet_encoding.py
from test_framework import generic_test
import string
import functools


def ss_decode_col_id(col: str) -> int:
    # n = 0
    # for i in range(len(col)):
    #     n += (string.ascii_uppercase.index(col[~i]) + 1) * (26 ** i)
    return functools.reduce(
        lambda result, d: (string.ascii_uppercase.index(d) + 1) + result * 26, col, 0
    )


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('spreadsheet_encoding.py',
                                       'spreadsheet_encoding.tsv',
                                       ss_decode_col_id))

File: epi_judge_cpp/spreadsheet_encoding.cc
#include <string>

#include "test_framework/generic_test.h"
using std::string;

int SSDecodeColID(const string &col) {
  int encoded{0};
  for (auto &ch : col) {
    encoded *= 26;
    encoded += ch - 'A' + 1;
  }
  return encoded;
}

int main(int argc, char *argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"col"};
  return GenericTestMain(args, "spreadsheet_encoding.cc",
                         "spreadsheet_encoding.tsv", &SSDecodeColID,
                         DefaultComparator{}, param_names);
}
